aten/src/ATen/native/quantized/cudnn/conv_prepack.cpp

```diff
@@ -72,6 +74,21 @@ c10::intrusive_ptr<ConvPackedParamsBase<kSpatialDim>> PackedConvWeightCudnn<
     // bias_contig = bias->contiguous();
   }
 
+  // cudnn v8.4.0 expects conv2d's weight tensor's input and output channels to be a multiple of 4. if it is not
+  // we need to explicitly pad it to a multiple of 4 ourselves as cudnn does not currently support padding.
+  // TODO: when and if cudnn enables padding in their operators, we can remove padding on our end;
+  // currently, limit padding support to groups=1 (ungrouped conv)
+  // TODO: implement this for groups > 1
+  auto num_output_channels = weight.size(0);
+  if (weight.size(0) % 4 != 0 || weight.size(1) % 4 != 0) {
+    int8_t nslices2pad_out = (4 - weight.size(0) % 4) % 4; // number of slices we need to pad/add for output channels
```

Comment:
can this be simplified to (4 - weight.size(0) % 4)? (removing the extra %4 in the end)

Reply:
not with the way the conditional in the if statement is set up because it's possible for weight.size(0) % 4 to be 0, but I can rewrite it in a better way